include/dali/imgcodec/image_decoder_interfaces.h

```diff
@@ -75,10 +75,35 @@ struct ROI {
   }
 };
 
-struct DecodeResult {
-  bool success;
-  std::exception_ptr exception;
-};
+template <typename OutShape>
+void OutputShape(OutShape &out_shape,
+                 const ImageInfo &info, const DecodeParams &params, const ROI &roi) {
+  int ndim = info.shape.sample_dim();
+  resize_if_possible(out_shape, ndim);
+  assert(size(out_shape) == ndim);  // check the size, in case we couldn't resize
```

Comment:
If resize may go awry maybe this sould be DALI_ENFORCE?

Reply:
It can't - the point is, it's not a run-time condition - it's a coding error.
However, now that I think of it, perhaps the OutputShape shouldn't go to the interfaces at all and land in a separate header or the `image_decoder_impl.h`.